src/box/alter.cc

```diff
@@ -4048,7 +4048,7 @@ on_replace_dd_cluster(struct trigger *trigger, void *event)
 		 * case it will replay this operation during the final join
 		 * stage.
 		 */
-		if (!replicaset.is_joining && replica_id == instance_id) {
+		if (!box_is_configured() && replica_id == instance_id) {
```

Comment:
Thanks for noticing this, indeed that is a bug. I was wondering why didn't any test fail. And realized that this check is useless. We no longer have this problem. Rebootstrap/bootstrap fetch data directly from memory, not from a snapshot file. Thus the replica can't see deletions of itself during join/rejoin. Even if it was deleted from `_cluster` before that.

It can probably happen if master would delete the replica after the join/rejoin is started, but I figured people just shouldn't do that.

I made this check stricter - now deletions of self from `_cluster` are not allowed no matter when.

Reply:
Nice, thanks for digging into this!